### ISSUE TITLE
Pmd tasks block on lock, which prevents concurrent execution

### ISSUE THREAD

**Issue by AndreasTu** [gradle/gradle#10884]
Since commit c04965b24d786d0deb97c243386f1b789849c9c7 from @big-guy  "Workaround issues with PMD inspecting Gradle's classpath", the PMD task blocks on the Lock of SystemProperties.instance.withSystemProperty(). 
This breaks concurrent execution of PMD tasks of multiple Projects.

See class `PmdInvoker` line 53.

### Expected Behavior
The Pmd tasks shall execute the analysis concurrently.

### Current Behavior
Only one Pmd task is executed andd all other block on the lock and block all workers.

### Context
Introduced in v5.6.0-RC1

### Steps to Reproduce 
Create a multi project build with at least two projects which executes a Pmd Analysis.
The second Pmd task will block.

### Your Environment
Gradle 5.6.2
Windows

### Stacktrace
The blocking stacktrace is:
`org.gradle.internal.SystemProperties.withSystemProperty(String, String, Factory)
   org.gradle.internal.SystemProperties$withSystemProperty.call(Object, Object, Object, Object)	
      org.gradle.api.plugins.quality.internal.PmdInvoker.invoke(Pmd)	
         org.gradle.api.plugins.quality.Pmd.run()
`

**Comment by AndreasTu** [comment]
@big-guy could you please have a look at this. It has a big impact on our project, due to the performance degradation. Do you need further details?

**Comment by APaloski** [comment]
I'd like to second what Andreas said, we are also seeing a huge impact on build performance due to this.

**Comment by regrog** [comment]
@AndreasTu is this problem still present in grade 6+ ?
Are you running PMD analysis with `--parallel `?
I'm investigating why my builds are slow and need a lot of memory.

**Comment by AndreasTu** [comment]
@regrog Yes this issue is still open in Gradle 6.0.1 (maybe also later, not tested that).
Yes my gradle build is running with `org.gradle.parallel=true`.

A characteristic for this issue is, that Gradle displays many pmdMain tasks executing but the process explorer only shows one fully utilized CPU core.

**Comment by regrog** [comment]
I was thinking of opening an issue based on this comment https://github.com/pmd/pmd/issues/1379#issuecomment-430356254

**Comment by regrog** [comment]
> @regrog Yes this issue is still open in Gradle 6.0.1 (maybe also later, not tested that).
> Yes my gradle build is running with `org.gradle.parallel=true`.
> 
> A characteristic for this issue is, that Gradle displays many pmdMain tasks executing but the process explorer only shows one fully utilized CPU core.

I recently reported this: https://github.com/gradle/gradle/issues/13594

**Comment by big-guy** [comment]
To at least get people back to the old behavior, I've made some changes in the upcoming 6.7.

If you have incremental analysis disabled, it won't block in this way. To properly fix it, we're going to have to do some more work to move Pmd into a separate process since it's tied to reading the `java.class.path` system property to do proper incremental analysis.

**Comment by big-guy** [comment]
This is still an issue.

Now that we have an AntWorkAction (used by Checkstyle), it should be easier to move PMD into a separate process.
